### DISCUSSION TITLE
Using pgmodeler on database where populated tables already exist

### DISCUSSION THREAD

**Discussion by hg2581** [pgmodeler/pgmodeler#1517] [Q&A]
I would like to use pgmodeler to expand an existing database which right now contains three tables that are fully populated. Were these tables to be deleted accidentally by me - perhaps by deleting the database in pgmodeler - it would take up to three weeks to repopulate them so this is something I am very leery of.

What would be a good way to set up this project in pgmodeler? Ideally making it /impossible/ for me to accidentally delete existing tables or the existing database while allowing me to expand it with more tables that are also using the other tables as keys?

**Comment by rkhaotix** [comment]
First and more important: always have a reliable backup policy. Accidents may occur occasionally. 

Second: you can use the diff tool to construct incrementally your database. I would start making a reverse engineering of your current database, perform changes over the model, and then generating diff scripts from that model.

Notice that there are situations in the diff that may cause the recreation of some objects which means a drop followed by a create command, so I would also test applying diffs in a development environment first before applying in production.

**Reply by hg2581** [reply]
The database is around 500 GB today so backups are pretty large... Maybe I should create another database just for the development with copies of the table structures but without any of the live data and then do my development in that environment? Once it is "finished" I could copy the new tables etc. into the live database structure?

**Reply by rkhaotix** [reply]
Exactly. I don't recommend applying diff scripts directly on production environments.
